internal/cache/metadata/stat_cache.go

```diff
@@ -176,30 +201,86 @@ func (sc *statCacheBucketView) AddNegativeEntry(objectName string, expiration ti
 	}
 }
 
+func (sc *statCacheBucketView) AddNegativeEntryForFolder(folderName string, expiration time.Time) {
+	name := sc.key(folderName)
+
+	// Insert a negative entry.
+	e := entry{
+		f:          nil,
+		expiration: expiration,
+		key:        name,
+	}
+
+	if _, err := sc.sharedCache.Insert(name, e); err != nil {
+		panic(err)
+	}
+}
+
 func (sc *statCacheBucketView) Erase(objectName string) {
 	name := sc.key(objectName)
 	sc.sharedCache.Erase(name)
 }
 
 func (sc *statCacheBucketView) LookUp(
 	objectName string,
-	now time.Time) (hit bool, m *gcs.MinObject) {
+	now time.Time) (bool, *gcs.MinObject) {
 	// Look up in the LRU cache.
-	value := sc.sharedCache.LookUp(sc.key(objectName))
+	hit, entry := sc.sharedcachelookup(objectName, now)
+	if hit {
+		return hit, entry.m
+	}
+
+	return false, nil
```

Comment:
Isn't the hit false here. It will be easy when you use named parameters for return types.
func (sc *statCacheBucketView) LookUp(
	objectName string,
	now time.Time) (hit bool, m *gcs.MinObject) {
hit, entry := sc.sharedCacheLookup(objectName, now)
if hit {
   m = entry.m
}

return

Reply:
line 230 hit is true and hit (true) is returned and line no 233 it should be false, seems working as expected, could please point which line no is returning incorrect value. Also should we avoid naked returns in all methods to be consistent, please see [ref](https://go.dev/tour/basics/7#:~:text=They%20can%20harm%20readability%20in%20longer%20functions.)